### PR TITLE
feat(e2e-runner): Docker image for running workflow-e2e tests

### PR DESCRIPTION
**Pull request by jchauncey**
- Lease k8s cluster using k8s-claimer
- Update helm cache
- Checkout
- Clean cluster if needed (uninstall workflow)
- Install workflow
- Health check deis until all components are up
- git checkout
- Install workflow-e2e chart
- tail log test container log file
- delete lease

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+FROM ubuntu:16.04
```
Comment (vdice) [inline Dockerfile]: total curiosity here - could we base this off of our ubuntu slim-based [docker-base img](https://github.com/deis/docker-base)? for our purposes here, we'd really only get `curl` and `bash`... but maybe it'd reduce img size?

File: README.md [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
+
+A few different things are happening here. First we export all the local environment variables to a file so we can pass those into the container at runtime. This allows us to use secret text values for things like `$AUTH_TOKEN`. When we call `docker run` we make sure we specify the user `jenkins:jenkins`. This way the container has write permissions into the volume mount.
 
+## Running the tests locally
+```
+$ docker run -e AUTH_TOKEN=$AUTH_TOKEN quay.io/deisci/e2e-runner
+```
+
+## Environment Variables
+* `AUTH_TOKEN` - Token needed to talk to k8s claimer
```
Comment (vdice) [inline README.md]: could we link to [k8s-claimer](https://github.com/deis/k8s-claimer) here (or somewhere in this file when talkin' bout k8s-claimer stuff)

Comment (jchauncey) [inline README.md]: yup

File: Dockerfile (hunk 1 above)
Comment (jchauncey) [inline Dockerfile]: yeah he hadnt published the image when i was working on this. but its updated now

**Comment by bacongobbler** [comment]
My only suggestion would be to move `scripts/` to `rootfs/home/jenkins` so the COPY operation can be a dumb `COPY rootfs /`. Otherwise, LGTM.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	jenkins
+
+RUN curl -s https://get.helm.sh | bash && \
+    mv helmc /usr/bin
+
+RUN curl -Ls https://bintray.com/deis/deisci/download_file?file_path=k8s-claimer-cli_git-8f0ddfd_linux-amd64 \
+      > /usr/bin/k8s-claimer && \
+    chmod +x /usr/bin/k8s-claimer
+
+RUN curl -Ls https://storage.googleapis.com/kubernetes-release/release/v1.2.3/bin/linux/amd64/kubectl \
```
Comment (bacongobbler) [inline Dockerfile]: There is a 1.2.4 release now. Might be worth just bumping this up

File: scripts/lease.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+lease() {
+  eval $(k8s-claimer --server=k8s-claimer-e2e.deis.com lease create --duration=$CLUSTER_DURATION)
+  echo "Leased cluster $CLUSTER_NAME for $CLUSTER_DURATION seconds"
+  echo "TOKEN: $TOKEN"
+  if [ -z $TOKEN ]; then
+    echo "Lease failed exiting."
+    exit
+  fi
+  # We should exit if we cant get a cluster
```
Comment (vdice) [inline scripts/lease.sh]: does this exit with non-zero code? (what we want, I would imagine)

Comment (bacongobbler) [inline scripts/lease.sh]: I'm pretty sure `exit` returns the last command's status code when no arguments are supplied, according to [this](http://tldp.org/LDP/abs/html/exit-status.html), which in this case would exit 0 unless `echo` fails. Good catch.

Comment (jchauncey) [inline scripts/lease.sh]: yeah oversight fix incoming

**Pull request merged**